File: nitro_modules/matiks-button/android/src/main/java/com/margelo/nitro/matiksbutton/HybridButtonView.kt
package com.margelo.nitro.matiksbutton

import android.content.Context
import android.view.View
import android.widget.Button
import com.margelo.nitro.matiksbutton.HybridButtonViewSpec

class HybridButtonView(
  context: Context
) : HybridButtonViewSpec() {

  private val button = Button(context)

  // Props
  override var title: String = ""
    set(value) {
      field = value
      button.text = value
    }

  override var onPress: () -> Unit = {}

  override val view: View
    get() = button

  init {
    button.setOnClickListener {
      onPress.invoke()
    }
  }
}

File: nitro_modules/matiks-button/ios/HybridButtonView.swift
import UIKit
import NitroModules

class HybridButtonView: HybridButtonViewSpec {

  private let button = UIButton(type: .system)

  // Props
  public var title: String = "" {
    didSet {
      button.setTitle(title, for: .normal)
    }
  }

  public var onPress: () -> Void = {}

  // Required by Nitro
  public var view: UIView {
    return button
  }

  public override init() {
    super.init()

    button.addTarget(
      self,
      action: #selector(handlePress),
      for: .touchUpInside
    )
  }

  @objc private func handlePress() {
    onPress()
  }
}
